app/views/interactive_pages/edit.html.haml

```diff
@@ -82,72 +82,25 @@
     - # TODO: Display preview of student completion page.
     =render :partial => "edit_completion"
   - else
-    - if @page.show_introduction
-      = text_editor @page, :text, { header: 'Introduction' }
+    - if @page.show_header
+      %div{:class => 'other full-width'}
+        #header_block.questions
+          %h2 Header Section
+          = render :partial => "edit_section", :locals => { :page => @page, :section_name => 'HEADER_BLOCK', :toggle_option => false }
     .text
       - if @page.show_info_assessment
         #info_assess_block.questions
           %h2 Info/Assess
-          %button.show-hide-toggle
-            Close All
-          .column-controls
-            = form_for @page do |f|
-              - if @page.has_attribute?('toggle_info_assessment')
-                %label
-                  = f.check_box :toggle_info_assessment, :onchange => 'this.form.submit();'
-                  Collapsible
-            = form_tag add_embeddable_activity_page_path(@activity, @page), :class => 'embeddables-form' do
-              .select_container
-                = embeddable_selector
-              = hidden_field_tag :approved_script_id
-              = hidden_field_tag :component_label
-              = submit_tag 'Add', {id: 'add_embeddable_button'}
-          .embeddables_list.sortable_embeddables
-            - @page.section_embeddables(nil).each do |e|
-              - if e.show_in_edit?
-                - tracked = e.respond_to?(:tracked_question) && e.tracked_question
-                - tracked_class = tracked ? 'tracked' : ''
-                .authorable.accordion_embeddable{:id => "embeddable_#{e.id}.#{e.class.to_s}", :class => tracked_class}
-                  %h3.embeddable_heading
-                    %span.embeddable_title
-                      - if e.respond_to? :name
-                        - if !e.name.nil? && !e.name.empty?
-                          #{e.name}
-                        - else
-                          #{e.class.model_name.human.titleize}
-                      - else
-                        #{e.class.model_name.human.titleize}
-                    %span.sortable-icon.ui-icon.ui-icon-arrowthick-2-n-s
-                  = render :partial => "#{e.class.name.underscore.pluralize}/author", :locals => { :embeddable => e, :page => @page, :allow_hide => true }
+          = render :partial => "edit_section", :locals => { :page => @page, :section_name => '', :toggle_option => true }
```

Comment:
Ah so section_name isn't nil. Unless it causes problems using nil here would be better so it matches how things are modeled.